unitTests/rest/SearchEndpointTest.ts

```diff
@@ -310,6 +311,48 @@ export function SearchEndpointTest() {
           });
         });
 
+        it('for plan', done => {
+          const qbuilder = new QueryBuilder();
+          qbuilder.expression.add('batman');
+          qbuilder.numberOfResults = 153;
+          qbuilder.enableCollaborativeRating = true;
+          const promiseSuccess = ep.plan(qbuilder.build());
+
+          expect(jasmine.Ajax.requests.mostRecent().url).toContain(ep.getBaseUri() + '/plan?');
+          expect(jasmine.Ajax.requests.mostRecent().url).toContain('organizationId=myOrgId');
+          expect(jasmine.Ajax.requests.mostRecent().url).toContain('potatoe=mashed');
+          expect(JSON.parse(jasmine.Ajax.requests.mostRecent().params)).toEqual(
+            jasmine.objectContaining({
+              q: 'batman',
+              numberOfResults: 153,
+              enableCollaborativeRating: true,
+              actionsHistory: []
+            })
+          );
+          expect(jasmine.Ajax.requests.mostRecent().method).toBe('POST');
+          expect(jasmine.Ajax.requests.mostRecent().requestHeaders).toEqual(
+            jasmine.objectContaining({
+              'Content-Type': 'application/json; charset="UTF-8"'
+            })
+          );
+
+          promiseSuccess
+            .then((data: IPlanResults) => {
+              expect(jasmine.Ajax.requests.mostRecent().responseType).toBe('json');
+            })
+            .catch((e: IErrorResponse) => {
+              fail(e);
+              return e;
+            })
+            .then(() => done());
+
+          jasmine.Ajax.requests.mostRecent().respondWith({
+            status: 200,
+            responseText: JSON.stringify({}),
```

Comment:
I was about to write a comment that we should simulate with an "actual" mock of a real response, but that's not really needed here for this test.

I guess, when we want to test the StandaloneComponent, we'll instead want to add a `Simulate.plan()` in the Simulate test file.